### PR TITLE
Add remaining API methods to auth-next

### PR DESCRIPTION
**Pull request by avolkovi**
Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

File: packages-exp/auth-exp/src/api/authentication/email_and_password.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+export enum GetOobCodeRequestType {
+  PASSWORD_RESET = 'PASSWORD_RESET',
+  EMAIL_SIGNIN = 'EMAIL_SIGNIN',
+  VERIFY_EMAIL = 'VERIFY_EMAIL',
+  VERIFY_AND_CHANGE_EMAIL = 'VERIFY_AND_CHANGE_EMAIL'
+}
+
+interface GetOobCodeRequest {
+  email?: string; // Everything except VERIFY_AND_CHANGE_EMAIL
+  // captchaResp?: string, // RESET_PASSWORD
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/api/authentication/email_and_password.ts:57]: Why are these commented out?

Comment (avolkovi) [reply packages-exp/auth-exp/src/api/authentication/email_and_password.ts:57]: they were commented out in the auth-ts branch, let me clean it up now that we know the signature better

File: packages-exp/auth-exp/src/api/index.ts [hunk 2]
```diff
@@ -62,17 +62,33 @@ export async function performApiRequest<T, V>(
 ): Promise<V> {
   const errorMap = { ...SERVER_ERROR_MAP, ...customErrorMap };
   try {
-    const body = request
-      ? {
+    let body = {};
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/api/index.ts:65]: @samhorlbeck this is now different from auth-ts due to GET param handling

File: packages-exp/auth-exp/src/api/authentication/email_and_password.ts (hunk 1 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/api/authentication/email_and_password.ts]: should be fixed now

File: packages-exp/auth-exp/src/api/index.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/api/index.ts:65]: I'm surprised this compiles. `body` is implicitly typed as an empty object, but later you assign a non-empty object

File: packages-exp/auth-exp/src/api/index.ts [hunk 3]
```diff
@@ -62,17 +62,33 @@ export async function performApiRequest<T, V>(
...
           body: JSON.stringify(request)
-        }
-      : {};
+        };
+      }
+    }
+
+    const queryString = Object.keys(params)
+      .map(
+        key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/api/index.ts:81]: nit: consider turning this into a standard closure to fix the formatting
```
.map(key => {
  return ...;
}).join('&')
```

File: packages-exp/auth-exp/src/api/index.ts (hunk 2 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/api/index.ts:65]: I think it's just typed as `object`

File: packages-exp/auth-exp/src/api/index.ts (hunk 3 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/api/index.ts:81]: fixed

**Pull request merged**